backend/app/assets/javascripts/spree/backend/views/quick_switch.js

```diff
@@ -0,0 +1,71 @@
+Spree.Views.QuickSwitch = Backbone.View.extend({
+  events: {
+    "shown.bs.modal": "onShow",
+    "hidden.bs.modal": "onHide",
+    "submit form": "onSubmit"
+  },
+
+  initialize: function() {
+    this.$el = $(this.el);
+    this.$dialog = $("[data-js='quick-switch-dialog']", this.$el);
+    this.$form = $("form", this.$el);
+    this.$input = $("[data-js='quick-switch-input']", this.$el);
+    this.$tooltip = $("[data-js='quick-switch-tooltip']", this.$el);
+		this.render();
+  },
+
+  triggerShortcut: function() {
+    this.onShortcut();
+  },
+
+  onShortcut: function() {
+    this.$el.modal("toggle");
+  },
+
+  onShow: function() {
+    this.$input.focus();
+  },
+
+  onHide: function() {
+    this.$input.val("");
+    $(".alert", this.$dialog).remove();
+  },
+
+  onSubmit: function(e) {
+    e.preventDefault();
+    this.toggleSearchClass();
```

Comment:
Is there a circumstance where you would want `onSubmit` to _remove_ the search class? You might want to explicitly enable and disable instead of relying on implicit toggle.

Reply:
In its current state there is no case where it needs this but that can certainly come if this ever gets more complicated.